kernel/uGraph.ml

```diff
@@ -219,10 +222,9 @@ let get_algebraic = function
 | Set -> Universe.type0
 | Type u | QSort (_, u) -> u
 
-let check_eq_sort ugraph s1 s2 = match s1, s2 with
+let check_eq_sort ugraph s1 s2 = type_in_type ugraph || match s1, s2 with
 | (SProp, SProp) | (Prop, Prop) | (Set, Set) -> true
-| (SProp, _) | (_, SProp) | (Prop, _) | (_, Prop) ->
-  type_in_type ugraph
```

Comment:
Note that in this line we are already unify different sorts in the current system